File: CCIP/CCIP/Tabs/MoreTab/TicketView/MyTicketViewController.m
//
//  MyTicketViewController.m
//  CCIP
//
//  Created by 腹黒い茶 on 2017/07/24.
//  Copyright © 2017年 CPRTeam. All rights reserved.
//

#import "MyTicketViewController.h"
#import "AppDelegate.h"

@interface MyTicketViewController ()

@end

@implementation MyTicketViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    NSString *noticeText;
    if ([AppDelegate haveAccessToken]) {
        CGImageRef QRImage = [TicketQRCodeImage generate:[AppDelegate accessToken]
                                                    size:self.ivQRCode.frame.size
                                         backgroundColor:[CIColor colorWithCGColor:[[UIColor whiteColor] CGColor]]
                                         foregroundColor:[CIColor colorWithCGColor:[[UIColor blackColor] CGColor]]
                                               watermark:nil];
        UIImage *qrImage = [UIImage imageWithCGImage:QRImage];
        [self.ivQRCode setImage:qrImage];
        noticeText = NSLocalizedString(@"TicketNotice", nil);
    } else {
        noticeText = NSLocalizedString(@"TicketNonExistNotice", nil);
    }
    [self.lbNotice setText:noticeText];
    [self.lbNotice setTextColor:[AppDelegate AppConfigColor:@"CardTextColor"]];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: CCIP/CCIP/Tabs/MoreTab/TicketView/MyTicketViewController.swift
//
//  MyTicketViewController.swift
//  OPass
//
//  Created by 腹黒い茶 on 2019/2/8.
//  Copyright © 2019 OPass. All rights reserved.
//

import Foundation
import UIKit

@objc class MyTicketViewController : UIViewController {
    @IBOutlet var lbNotice: UILabel?
    @IBOutlet var ivQRCode: UIImageView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let noticeText: String
        if (AppDelegate.haveAccessToken()) {
            let QRImage = TicketQRCodeImage.generate(
                AppDelegate.accessToken(),
                size: (self.ivQRCode?.frame.size)!,
                backgroundColor: CIColor.white,
                foregroundColor: CIColor.black,
                watermark: nil
            )
            let qrImage = UIImage.init(cgImage: QRImage!)
            self.ivQRCode?.image = qrImage;
            noticeText = NSLocalizedString("TicketNotice", comment: "")
        } else {
            noticeText = NSLocalizedString("TicketNonExistNotice", comment: "");
        }
        self.lbNotice!.text = noticeText
        self.lbNotice?.textColor = AppDelegate.appConfigColor("CardTextColor")
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
